src/main/java/org/terasology/module/health/core/CoreHealthSystem.java

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.module.health.core;
```

Comment:
I've put this in a separate package, although I'm not sure about it. My goal is to separate the API of Health other modules can and should integrate with from the _default logic_ served by this module. 

Other aspects that I see in this system are the short timeout before regeneration when an entity is hit. These default feature themselves use the API and serve as example implementations (at least that's my idea - "eat your own dog food" :dog: )